### PR TITLE
[qa] Overhaul p2p_compactblocks.py

### PR DESCRIPTION
**Pull request by sdaftuar**
Remove tests of:
 - compactblock behavior in a simulated pre-segwit version of bitcoind
   This should have been removed a long time ago, as it is not generally
   necessary for us to test the behavior of old nodes (except perhaps if we
   want to test that upgrading from an old node to a new one behaves properly)

 - compactblock behavior during segwit upgrade (ie verifying that network
   behavior before and after activation was as expected)
   This is unnecessary to test now that segwit activation has already happened.

### CODE REVIEW

**Comment by sdaftuar** [comment]
This makes changes like that proposed in #15633 much easier to make (I have a relatively small diff to the test that would support that change in particular, should we want to adopt a different behavior).

File: test/functional/p2p_compactblocks.py [hunk 1]
```diff
@@ -135,11 +131,18 @@ def make_utxos(self):
...
         block2.hashMerkleRoot = block2.calc_merkle_root()
         block2.solve()
-        self.test_node.send_and_ping(msg_block(block2))
+        self.segwit_node.send_and_ping(msg_block(block2))
         assert_equal(int(self.nodes[0].getbestblockhash(), 16), block2.sha256)
         self.utxos.extend([[tx.sha256, i, out_value] for i in range(10)])
         return
 
+    # Send balance to a segwit output
+    def make_segwit_output(self, node):
```
Comment (practicalswift) [inline test/functional/p2p_compactblocks.py]: Nit: Since `self` is not used -- this could be a function instead of a method?

**Review by jnewbery — COMMENTED** [review]
utACK 5bc755fdd166e84d8606a613c379705b362f0bf2

I've left some comments inline. One is a bug while the rest are style nits.

Whilst reviewing this, a lot of tidy-ups suggested themselves to me, so I went ahead and made a branch here: https://github.com/jnewbery/bitcoin/tree/tidy_up_p2p_compactblocks. I think the result is quite a bit clearer and easier to follow than the existing p2p_compactblocks.py. I've split out the commits in a vaguely logical way so it should be pretty easy to review. Feel free to take any or all of those changes, or we can follow this up with another PR.

File: test/functional/p2p_compactblocks.py [hunk 2]
```diff
@@ -94,11 +93,9 @@ def send_await_disconnect(self, message, timeout=30):
 class CompactBlocksTest(BitcoinTestFramework):
     def set_test_params(self):
         self.setup_clean_chain = True
-        # Node0 = pre-segwit, node1 = segwit-aware
-        self.num_nodes = 2
+        self.num_nodes = 1
         # This test was written assuming SegWit is activated using BIP9 at height 432 (3x confirmation window).
```
[the remark this answers is not in the document]
Comment (sdaftuar) [reply test/functional/p2p_compactblocks.py]: Done.

File: test/functional/p2p_compactblocks.py [hunk 3]
```diff
@@ -135,11 +131,18 @@ def make_utxos(self):
...
+        self.segwit_node.send_and_ping(msg_block(block2))
         assert_equal(int(self.nodes[0].getbestblockhash(), 16), block2.sha256)
         self.utxos.extend([[tx.sha256, i, out_value] for i in range(10)])
         return
 
+    # Send balance to a segwit output
+    def make_segwit_output(self, node):
+        if node.getbalance() < 2:
+            node.generate(101)
+        address = node.getnewaddress("bech32")
```
[the remark this answers is not in the document]
Comment (sdaftuar) [reply test/functional/p2p_compactblocks.py]: Fixed.

File: test/functional/p2p_compactblocks.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (sdaftuar) [reply test/functional/p2p_compactblocks.py]: Done.

File: test/functional/p2p_compactblocks.py [hunk 4]
```diff
@@ -793,126 +785,56 @@ def announce_cmpct_block(node, peer):
...
 
-        self.log.info("Running tests, pre-segwit activation:")
+        assert_equal(get_bip9_status(self.nodes[0], "segwit")["status"], 'active')
 
         self.log.info("Testing SENDCMPCT p2p message... ")
-        self.test_sendcmpct(self.nodes[0], self.test_node, 1)
-        sync_blocks(self.nodes)
-        self.test_sendcmpct(self.nodes[1], self.segwit_node, 2, old_node=self.old_node)
-        sync_blocks(self.nodes)
+        self.test_sendcmpct(self.nodes[0], self.segwit_node, 2, old_node=self.old_node)
```
[the remark this answers is not in the document]
Comment (sdaftuar) [reply test/functional/p2p_compactblocks.py]: Done.

[the remark this answers is not in the document]
Comment (sdaftuar) [reply test/functional/p2p_compactblocks.py]: Simplified.

File: test/functional/p2p_compactblocks.py [hunk 5]
```diff
@@ -793,126 +785,56 @@ def announce_cmpct_block(node, peer):
 
     def run_test(self):
         # Setup the p2p connections
-        self.test_node = self.nodes[0].add_p2p_connection(TestP2PConn())
-        self.segwit_node = self.nodes[1].add_p2p_connection(TestP2PConn(), services=NODE_NETWORK | NODE_WITNESS)
-        self.old_node = self.nodes[1].add_p2p_connection(TestP2PConn(), services=NODE_NETWORK)
+        self.segwit_node = self.nodes[0].add_p2p_connection(TestP2PConn(), services=NODE_NETWORK | NODE_WITNESS)
```
[the remark this answers is not in the document]
Comment (sdaftuar) [reply test/functional/p2p_compactblocks.py]: Removed.

File: test/functional/p2p_compactblocks.py (hunk 1 above)
Comment (sdaftuar) [reply test/functional/p2p_compactblocks.py]: Eliminated this in the latest version.

**Comment by sdaftuar** [comment]
I've taken a bunch of the feedback and cherry-picked a few commits from @jnewbery's branch, and updated this PR.  Old version is here: https://github.com/sdaftuar/bitcoin/commits/15660.1

**Comment by jnewbery** [comment]
tested ACK f97b82149a5cda1443f8f1d085b95c7987b9ba1f. Please squash commits.

**Comment by sdaftuar** [comment]
Squashed.

**Comment by jnewbery** [comment]
utACK 7813eb1db132c023902ad945995cc32a325893ca

Travis failure is unrelated. I've restarted it.

**Pull request merged**